createas.c

```diff
@@ -841,6 +893,39 @@ check_ivm_restriction_walker(Node *node, check_ivm_restriction_context *context)
 
 				query_tree_walker(qry, check_ivm_restriction_walker, (void *) context, QTW_IGNORE_RT_SUBQUERIES);
 
+				/* additional restriction checks for exists subquery */
```

Comment:
Could you make  some comments for explaining why this additional check is required?

Reply:
Sure, I added some comments.